design-documents/graph-ql/coverage/negotiableQuotes.graphqls

```diff
@@ -0,0 +1,361 @@
+
+    createOrderWithPaymentInfoNegotiableQuote(
+        paymentInformation: NegotiableQuotePaymentInformationInput!
+    ): CreateOrderWithPaymentInfoNegotiableQuoteOutput! @doc(description:"Assigns payment information and the billing address for the negotiable quote to create order")
+
+    createOrderWithGiftCardForNegotiableQuote(
+        giftCardAccountData: NegotiableQuoteGiftCardAccountDataInput!
+    ): CreateOrderWithGiftCardForNegotiableQuoteOutput! @doc(description:"Assigns GiftCard information and cart ID for the negotiable quote to create order")
+
+    removeGiftCardFromNegotiableQuote(
+        giftCardsData: NegotiableQuotegiftCardsInput!
+    ): RemoveGiftCardFromNegotiableQuoteOutput! @doc(description:"Removes a gift card that has been applied to a negotiable quote")
+}
+
+type Customer {
+        negotiable_quotes(
+        page_size: Int = 20,
+        current_page: Int = 1,
+        filter: NegotiableQuoteFilterInput
+    ): [NegotiableQuote] @doc(description: "Get negotiable quotes of a customer")
+}
+
+type NegotiableQuote {
+    billing_address: BillingAddress! @doc(description: "Billing address of negotiable quote")
+    attachements: [AttachmentContent] @doc(description: "Negotiable quote Attachements")
+    comments: [NegotiableQuoteComment] @doc(description: "Negotiable quote comments")
+    totals: TotalInfo @doc(description: "Negotiable quote totals info.")
+    payment_info: NegotiableQuotePaymentInfo! @doc(description: "Negotiable quote payment info.")
+}
+
+input NegotiableQuoteFilterInput {
+    ids: FilterEqualTypeInput @doc(description: "Filter Customer Negotibale quotes with an negotiable quote ID or list of negotiable quote IDs")
+    name: FilterMatchTypeInput @doc(description: "Filter by display name of the negotiable quote")
+}
+
+type BillingAddress {
+    id: ID! @doc(description: "The ID of the billing address")
+    firstname: String @doc(description: "The first name of the person associated with the billing address")
+    lastname: String @doc(description: "The family name of the person associated with the billing address")
+    customer_id: String @doc(description: "The customer ID")
+    email: String @doc(description: "The customer email id")
+    company: String @doc(description: "The customer's company")
+    telephone: String @doc(description: "The telephone number")
+    street: [String] @doc(description: "An array of strings that define the street number and name")
+    city: String @doc(description: "The city or town")
+    region: String @doc(description: "The region name")
+    region_id: String @doc(description: "The region ID")
+    region_code: String @doc(description: "The region code")
+    postcode: String @doc(description: "The customer's ZIP or postal code")
+    country_id: String @doc(description: "Deprecated: use `country_code` instead.")
+    same_as_billing: Int @doc(description: "Billing address confirmation")
```

Comment:
Why does the `BillingAddress` type have a `same_as_billing` field? Seems circular

This might be some weirdness in the Magento data model I don't know about, but GraphQL gives us a new opportunity to rethink things, if that's the case.

Reply:
In  NQ REST API, the BillingAddress takes AddressInterface as one parameter which includes same_as_billing.

Reply:
We're not looking to copy the REST API here luckily! If some pattern seems incorrect, we shouldn't copy it.

Reply:
There is a comment which suggests us to use `CartAddressInterface` instead of `BillingAddress` https://github.com/magento/architecture/pull/381#discussion_r431472087, but we are having few concerns which are mentioned in this https://github.com/magento/architecture/pull/381#discussion_r432461251